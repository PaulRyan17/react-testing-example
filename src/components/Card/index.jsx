import React from "react";

const Card = ({ handleClick }) => {
  return (
    <div className="card">
      <div className="text">
        <div className="action">
          <span>VISIT</span>
        </div>
        <div className="country">
          <span>Iceland</span>
        </div>
      </div>
      <button onClick={handleClick}>DISCOVER</button>
    </div>
  );
};

export default Card;
