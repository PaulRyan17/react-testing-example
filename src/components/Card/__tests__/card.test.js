import React from "react";
import Card from "..";
import Renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme'
configure({adapter: new Adapter()});

it("renders correctly", () => {
  const tree = Renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('triggers the callback function', ()=>{
  // in order to test the callback function we first need to mock it
  const onClickCallback = jest.fn()
  // pass our mock function as the prop the Card
  let card = mount(<Card handleClick={onClickCallback} />)
  // find out button i.e. the button that triggers the callback and click it
  const button = card.find('button').simulate('click')
  // our mock function should have been called once
  expect(onClickCallback).toHaveBeenCalledTimes(1)
})

it('callback function is the passed the string discover', ()=>{
  // in order to test the callback function we first need to mock it
  const onClickCallback = jest.fn()
  // pass our mock function as the prop the Card
  let card = mount(<Card handleClick={onClickCallback} />)
  // find out button i.e. the button that triggers the callback and click it
  const button = card.find('button').simulate('click')
  // our mock function should have been called once
  expect(onClickCallback).toHaveBeenCalledWith('discover')
})

