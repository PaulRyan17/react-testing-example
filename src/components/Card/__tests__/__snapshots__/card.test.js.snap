exports[`test renders correctly 1`] = `
<div
  className="card">
  <div
    className="text">
    <div
      className="action">
      <span>
        VISIT
      </span>
    </div>
    <div
      className="country">
      <span>
        Iceland
      </span>
    </div>
  </div>
  <button
    onClick={[Function]}>
    DISCOVER
  </button>
</div>
`;
